src/diagonal.works/b6/ingest/mutable.go

```diff
@@ -862,32 +860,36 @@ func (m *MutableOverlayWorld) HasFeatureWithID(id b6.FeatureID) bool {
 }
 
 func (m *MutableOverlayWorld) findFeatureByID(id b6.FeatureID) b6.Feature {
+	var feature Feature
 	switch id.Type {
 	case b6.FeatureTypePoint:
 		if p, ok := m.byID.Points[id.ToPointID()]; ok {
-			return newPointFeature(p)
+			feature = p
 		}
 	case b6.FeatureTypePath:
 		if p, ok := m.byID.Paths[id.ToPathID()]; ok {
-			return newPathFeature(p, m)
+			feature = p
 		}
 	case b6.FeatureTypeArea:
 		if a, ok := m.byID.Areas[id.ToAreaID()]; ok {
-			return newAreaFeature(a, m)
+			feature = a
 		}
 	case b6.FeatureTypeRelation:
 		if r, ok := m.byID.Relations[id.ToRelationID()]; ok {
-			return newRelationFeature(r, m)
+			feature = r
 		}
 	case b6.FeatureTypeCollection:
 		if c, ok := m.byID.Collections[id.ToCollectionID()]; ok {
-			return newCollectionFeature(c, m)
+			feature = c
 		}
 	case b6.FeatureTypeExpression:
 		if e, ok := m.byID.Expressions[id.ToExpressionID()]; ok {
```

Comment:
random but i was thinking before when doing collections if we need to have separate maps for each feature type b/c we index by id anyways? / n/a to this cl anyways

Reply:
It was really just for type safety, because the maps can be map[PointID]*PointFeature, otherwise we'd need to do map[FeatureID]Feature, and then downcast everywhere, which to be honest might not be such a big deal, because we do that a lot. It would also make the key-size of the map bigger too (because we'd have to include the feature type).

Reply:
yea no i was thinking precisely keeping it at feature level could simplify some stuff, like the lookups, merging and copying, and so on, we wouldnt necessarily need to reimplement all the interfaces. i imagine we d only have to know the type in certain places, a lot of what i was adding for collection seemed like it could ve been done on feature. fair enough re key but that s like 10 characters bigger tops, not really relevant? i might look into this just for fun at some point then, when avoiding other life stuff

Reply:
it would also potentially make adding new features much easier (not that it was hard but just like less stuff to update), which was my primary motivation while having these thoughts